cmd/util/app.go

```diff
@@ -648,3 +650,50 @@ func setAnnotations(app *argoappv1.Application, annotations []string) {
 		}
 	}
 }
+
+// liveObjects deserializes the list of live states into unstructured objects
+func LiveObjects(resources []*argoappv1.ResourceDiff) ([]*unstructured.Unstructured, error) {
+	objs := make([]*unstructured.Unstructured, len(resources))
+	for i, resState := range resources {
+		obj, err := resState.LiveObject()
+		if err != nil {
+			return nil, err
+		}
+		objs[i] = obj
+	}
+	return objs, nil
+}
+
+func FilterResources(groupChanged bool, resources []*argoappv1.ResourceDiff, group, kind, namespace, resourceName string, all bool) []*unstructured.Unstructured {
```

Comment:
As a util function, can this be modified to return errors instead of using `errors.CheckError` or `log.Fatal`? It should be up to the caller how to handle errors. 

This change will also make it possible to unit test the error conditions of this function.

Reply:
@pasha-codefresh if you keep pushing new commits to this PR (not rebasing it) it will be fairly easy to review the new code you push. Honestly I would prefer to have the unit tests as part of this PR.

Reply:
I dont mind, i will add them